test/gtest/ucp/test_ucp_am.cc

```diff
@@ -1055,6 +1055,23 @@ class test_ucp_am_nbx_rndv : public test_ucp_am_nbx {
         return UCS_INPROGRESS;
     }
 
+    static ucs_status_t am_data_wrong_rndv_cb(void *arg,
+                                              const void *header,
+                                              size_t header_length,
+                                              void *data, size_t length,
+                                              const ucp_am_recv_param_t *param)
+    {
+        void **args = reinterpret_cast<void**>(arg);
+        test_ucp_am_nbx_rndv *self = reinterpret_cast<test_ucp_am_nbx_rndv*>(args[0]);
+        void **data_desc_p = reinterpret_cast<void**>(args[1]);
+
+        *data_desc_p = data;
+        self->m_am_received = true;
+
+        /* return the wrong UCS_OK, instead of UCS_INPROGRESS */
```

Comment:
please mention that we drop the message, like
"return UCS_OK without calling ucp_am_recv_data_nbx to drop the message"